src/mito2/src/worker.rs

```diff
@@ -716,35 +720,17 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         sender: oneshot::Sender<SetReadonlyResponse>,
     ) {
         if let Some(region) = self.regions.get_region(region_id) {
-            region.set_writable(false);
+            // We need to do this in background as we need the manifest lock.
+            common_runtime::spawn_bg(async move {
```

Comment:
Are there any execution sequence issues if these kinds of tasks are put into background? Right now they are executed in order in worker loop, that means request B has to wait for request A if A arrives before B. However, background tasks do not have this guarantee, B could be executed before A.

Reply:
For requests (except flush and compaction) that might update the manifest, we set the state of the region in the worker loop before spawning a background job. The worker will return an error if a new request wants to modify the region state as `writable_region_or()` checks the region state.
https://github.com/GreptimeTeam/greptimedb/blob/314f2704d49d407e87152931b0050d677c8edc7a/src/mito2/src/region.rs#L235-L239

I also have a comment in https://github.com/GreptimeTeam/greptimedb/issues/3456#issuecomment-2049105369